### PR TITLE
Properly set failed expression evaluations to error state

### PR DESCRIPTION
**Pull request by hxiong388**
Properly set failed expression evaluations to a variable with error state instead of just error message String.

### CODE REVIEW

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint.rb [hunk 1]
```diff
@@ -260,22 +261,33 @@ def line
...
         ##
         # Evaluate the breakpoint's condition expression against a given binding
-        # object. Returns true if the condition expression evalutes to true;
-        # false if error happens or the expression evaluates to false.
+        # object. Returns true if the condition expression evalutes to true or
+        # there isn't a condition; otherwise false. Set breakpoint to error
+        # state if exception happens.
         #
         # @param [Binding] binding A Ruby Binding object
         # @return [Boolean] True if condition evalutes to true, false if
```
Comment (dazuma) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint.rb]: Update the return value documentation

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/status_message.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          ##
+          # Exports the StatusMessage to a
+          # Google::Devtools::Clouddebugger::V2::StatusMessage object.
+          def to_grpc
+            return nil if empty?
+            description_grpc =
+              Google::Devtools::Clouddebugger::V2::FormatMessage.new \
+                format: description.to_s
+
+            Google::Devtools::Clouddebugger::V2::StatusMessage.new(
```
Comment (dazuma) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/status_message.rb]: seattle-style nit: any reason you use parens here but no parens a few lines above?

Comment (quartzmo) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/status_message.rb]: Hmmm, I believe we have not been consistent on this, unfortunately.

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint.rb (hunk 1 above)
Comment (hxiong388) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint.rb]: fixed

File: google-cloud-debugger/lib/google/cloud/debugger/breakpoint/status_message.rb (hunk 2 above)
Comment (hxiong388) [inline google-cloud-debugger/lib/google/cloud/debugger/breakpoint/status_message.rb]: fixed

**Pull request merged**